### PR TITLE
Reducing StatsForecast Size

### PR DESCRIPTION
**Pull request by kvnkho**
There are some issues saying that the size of StatsForecast is too inconvenient for AWS Lambda. See [this](https://github.com/Nixtla/statsforecast/issues/596) and [this](https://github.com/Nixtla/statsforecast/issues/509). The package sits at 700 MB. 

We can check the size of the dependencies below. Note that Spark, Dask, and Ray are dev dependencies. Plotly (and Dash) are core dependencies.

Screen Shot 2023-08-04 at 12 03 38 AM

This pull request removes plotly as a core dependency in favor of matplotlib being the default.

The second thing is that the build includes all source notebooks (docs and source). The docs folder is around 70-75MB. Removing the docs will also save a lot of space, especially because the docs contain plotly images.

Note:

I had to change the test to pass. It has to do with Plotly and Matplotlib not having 1:1 parity with behavior. It's somewhat ok for now because the syntax will slightly differ, but users are unlikely to use both anyway. I can open another issue for that. 

EDIT: Docs are actually not included in the build already because there is no __init__.py . Action files are but they are so small anyway it doesn't matter.

### CODE REVIEW

File: statsforecast/core.py [hunk 1]
```diff
@@ -1686,6 +1693,33 @@ def plotly(
                 if n_cols == 1:
                     axes = np.array([axes])
 
+            def plotly(
```
Comment (jmoralez) [inline statsforecast/core.py:1696]: Why is this function being defined here?

File: setup.py [hunk 2]
```diff
@@ -45,7 +46,12 @@
     packages = setuptools.find_packages(),
     include_package_data = True,
     install_requires = requirements,
-    extras_require={'dev': dev_requirements, 'dask': dask_requirements, 'ray': ray_requirements, 'spark': spark_requirements,},
+    extras_require={'dev': dev_requirements, 
+                    'dask': dask_requirements, 
+                    'ray': ray_requirements, 
+                    'spark': spark_requirements,
+                    'plotly': plotly_requirements,
+                    },
```
Comment (jmoralez) [inline setup.py:54]:
```suggestion
    extras_require={
        'dev': dev_requirements,
        'dask': dask_requirements,
        'ray': ray_requirements,
        'spark': spark_requirements,
        'plotly': plotly_requirements,
    },
```

**Review by jmoralez — CHANGES_REQUESTED** [review]

File: statsforecast/core.py (hunk 1 above)
Comment (kvnkho) [reply statsforecast/core.py:1696]: I really don't know. The original code that had defined there, I tried not to move it.

Comment (kvnkho) [reply statsforecast/core.py:1696]: It's the equivalent here: https://github.com/Nixtla/statsforecast/blob/main/statsforecast/core.py#L1534

Comment (kvnkho) [reply statsforecast/core.py:1696]: Ah, I understand! Great catch. Will update.

Comment (kvnkho) [reply statsforecast/core.py]: I have updated it

**Review by jmoralez — APPROVED** [review]
LGTM. I'm adding the breaking label because if someone is relying on statsforecast installing plotly and having it as the default backend this will require them to add it as a dependency and setting the engine argument.

**Comment by kvnkho** [comment]
> LGTM. I'm adding the breaking label because if someone is relying on statsforecast installing plotly and having it as the default backend this will require them to add it as a dependency and setting the engine argument.

Yep that's right, unfortunately.

**Comment by DaveParr** [comment]
Looking forward to using this, thanks for the effort. 👍🏻

**Comment by nelsoncardenas** [comment]
Awesome job!

**Pull request merged**